src/util/os_helpers.go

```diff
@@ -17,6 +17,18 @@ func CreateEmptyDirectory(dir string) error {
 	return os.MkdirAll(dir, os.FileMode(0777))
 }
 
+// MakeDirectory creates a directory dir if it doesn't already exist, returns an error if any
+func MakeDirectory(dir string) error {
```

Comment:
Please make use of this function elsewhere to dry up the other uses of `os.MkdirAll`